artview/components/nav.py

```diff
@@ -258,40 +266,91 @@ def AdvanceFileSelect(self, findex):
             return
         self.fileindex = findex
         self.filename = self.Vfilelist.value[findex]
-        self.menu._openfile(filename=self.filename)
-        self._update_InfoUI()
+        self._openfile(self.filename)
 
     def goto_first_file(self):
         self.fileindex = 0
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
     def goto_last_file(self):
         self.fileindex = len(self.Vfilelist.value) - 1
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
     def goto_prev_file(self):
         self.fileindex = self.fileindex - 1
-        self.act_prev.setToolTip(
-            'Previous file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex - 1]))
-        self.act_next.setToolTip(
-            'Next file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex + 1]))
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
     def goto_next_file(self):
         self.fileindex = self.fileindex + 1
-        self.act_prev.setToolTip(
-            'Previous file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex - 1]))
-        self.act_next.setToolTip(
-            'Next file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex + 1]))
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
-    def NewFileList(self, variable, value, strong):
-        '''respond to change in radar.'''
-        if self.Vfilelist.value is None:
+    def _openfile(self, filename=None):
+        '''Open a file via a file selection window.'''
+        print("Opening file " + filename)
```

Comment:
Menu is such that opening a new file also update the filelist. We don't need this here. It look like the same function but it is not. And also we don't want a permanent connection with menu, otherwise we would not need shared variables.